### ISSUE TITLE
5 things that cause a serious problem in the operation of ZeroNet for several years

### ISSUE THREAD

**Issue by mx5kevin** [HelloZeroNet/ZeroNet#2602]
### Step 1: Please describe your environment

  * ZeroNet version: 0.7.1 (rev 4496) and all previous versions
  * Operating system: Windows 10
  * Web browser: Firefox 79.0
  * Tor status: not always
  * Opened port: no
  

### Describe the problem:

For simple users a few things are too unnecessarily complicated which should be simplified because this things make the software unusable for many user about long time. Setting up websites and downloading files should be made much easier for simple users.

### As website owner: 
Setting up websites should be made easier by copying and signing files to a folder.

-1.UTF-8 encoding: All pages must be converted to UTF-8 otherwise, the characters will not be displayed correctly in the browser. This requires a separate program for several files, which is difficult to find on the Internet.

-2.In content.json file, files larger than 10 mb are required as set as optional files or they cannot be downloaded. This is very inconvenient for site owners many times the larger files are part of the page. Not everyone can use the content.json commands. It would be simpler if no separate parameter is specified in content.json larger files would be downloaded in the same way as smaller files.

-3.Special line in HTML files: `<base href="" target="_top" id="base">` and `<meta charset="UTF-8">`
Don't have to setup the `<base href="" target="_top" id="base">` line to navigate correctly the pages. Many people don’t know to need to set it up and without it, you cannot navigate properly between pages.

### As user: 
Large files are not available on the network and this makes the use of the software unenjoyable. For simple users we should make it easy to download files. If it’s not even available right away when he visit back can check it out. There should be no complicated options for this because these cannot be handled by users. Most users cannot handle a +SEED button and all they see is that they can’t download anything.

-4.Large file download in single user zites: Download all large files automatically with the website. For simple users don’t have to worry about why the content (like larger fideo or etc files) isn’t available and why not. Advanced users should have a button if they do not want to download larger files on the site then this can be turned off at the touch of a button.

-5.Large file download in multiple user zites: If a user clicking a file always complete and resume the file download. If the file is not available in the default case,the program always try to download it later even if the computer restarts. If someone does not want this at the touch of a button, this can also be turned off. Semi-finished or finished files can be deleted at any time. I have been experiencing for years that that theory doesn’t work that we are not downloading the entire file from browser request so as not to take up extra disk space. As a result, the files cannot spread and almost nothing is available, and it is thus useless, unenjoyable to many people. Added to that is the problem some multiple user sites like IFS are not use the +SEED button.

**A small optimization would be important default setting ZeroNet run at system startup.To seed files this is very important.If not set the computer is active but users cannot seed the files.**

**Comment by slrslr** [comment]
> As user:
> -4.

this reminded me similar issue #2138 "Multifunctional corner button (easier access to site features & more seeding control for the user)"
Easier accessible control on what to download and upload would be good. User should be informed how much data and how many files it will attempt to DL.
Root problem is that ZN is limitted (OS, web browser - https://github.com/HelloZeroNet/ZeroNet/issues/2161 ?) in number of connections it can do thus not always effective in downloading too many optional files. Related issue "Page that shows how Zeronet is utilizing resource limits (connections/threads/sockets)"

> -5.
> files cannot spread and almost nothing is available, and it is thus useless, unenjoyable to many people

+1, see [this zerotalk topic](http://127.0.0.1:43110/1TaLkFrMwvbNsooF4ioKAY9EuxTBTjipT/?Topic:1571214527_1L4dZcDF2maSKHDy788yhxpYnBWnXadUtS)

**Comment by TrojkaDevil** [comment]
I agree with @mx5kevin in everything. ZeroNet sucks in its current form and with the presence of @shortcutme. Tamas is a Hungarian who don't speak even Hungarian properly and he was acknowledged publicly that TED denied him to give a speech and he needed to rewrite his speech (basically to turn against China) in order to give him the opportunity to give a speech in New York. If you ask me this is already laughable because he promotes ZeroNet as some censorship resistant network while he himself gives blowjob to the Americans and submitting himself for censorship by shitty conferences as TED. 

I think is a big waste of time to contribute to this repository since as you can guess he Tamas Kocsis will take all the credit.

**Comment by purplesyringa** [comment]
What's the point, antifa? You've been spamming us for over a year, yet no one listens to you. Stop this pointless discussion.

UPD: Perhaps the comment was deleted? Whatever, anyway

**Comment by mx5kevin** [comment]
For users we need to make the platform easy to use. There are many user who cannot handle a simple +SEED button in a zite. For website administrators in the same way, things need to be simplified. Copies the files to  the web page folder and signs them. In single user zites download all files what the page contains default way. In multi user zites resume and complete all clicked file download. These are the options where the user do not need any expertise to use the software. The program must also be set by default to run at system startup for seeding and downloading to be effective. It is an experience that there are many users who cannot handle even a little more complicated options. For them, a lot needs to be simplified and made more transparent things. Such as a MSG when the download is complete.

**Comment by ghost** [comment]
@mx5kevin @slrslr none of you should expect that @shortcutme will make this software entirely free (under GPLv3+) or that this repository going to be open for contributions! **Tamas Kocsis added a** **BACKDOOR to ZERONET** and when I found it @imachug accused me that I wanted to include the backdoor! See: https://github.com/HelloZeroNet/ZeroNet/issues/2606 and https://github.com/HelloZeroNet/ZeroNet/pull/2605

@imachug said:
> Hats off for a try to add a backdoor, but I am afraid you didn't manage to add it, "html" in content_type is enough. Try harder next time. https://github.com/HelloZeroNet/ZeroNet/pull/2605#issuecomment-674865654

**Comment by wandrien** [comment]
There seems to be several unrelated topics here. Better to make a separate issue for each.

> -4.Large file download in single user zites: Download all large files automatically with the website. For simple users don’t have to worry about why the content (like larger fideo or etc files) isn’t available and why not. Advanced users should have a button if they do not want to download larger files on the site then this can be turned off at the touch of a button.

It can be implemented on a site engine level, and in more sophisticated way than just simple downloading all the data.

For example, when you visit my blog, it starts downloading all the videos visible on the page.
Some users complained it's the wrong strategy and I shouldn't use their drive as a CDN without their consent.

**Comment by wandrien** [comment]
@slrslr 
> +1, see [this zerotalk topic](http://127.0.0.1:43110/1TaLkFrMwvbNsooF4ioKAY9EuxTBTjipT/?Topic:1571214527_1L4dZcDF2maSKHDy788yhxpYnBWnXadUtS)

There are two sides of the issue.

First, if ZeroNet works poorly on downloading big files, we should fix it. ZeroNet should resume downloading after restart, after computer wake up, when new peers go online etc.

Second, the culture. When a person publish a torrent, she knows she should stay online until several peers finish the download. And when we are in ZeroNet, we are probably used to the fact that everything works as a magic. Everything except big files. So we need some culture of sharing files in the same way as in BitTorrent.

**Comment by slrslr** [comment]
> When a person publish a torrent, she knows she should stay online until several peers finish the download. ... we need some culture of sharing files in the same way as in BitTorrent.

it may help if user is presented (on ZeroHello) with the list of 5 sites that contains least seeded files he shared. And zeronet notify user if he attempt to quit ZN. So user is aware about consequences and that he should start ZN ASAP else there will be problem. Or user is presented with option to remove/undo his content publishing for peerless files.

**Comment by leycec** [comment]
@slrslr has the right of it. Ideally, the ZeroHello frontpage would explicitly notify seeders of all big files for which they're the last remaining seeder – **in bold,** _with emphasis_, and possibly a GeoCities-era red blinking ComicSans font as well. That alone would do wonders for raising awareness amongst users of the need to continually seed big files (and thus run ZeroNet as a startup system service).
